addons/food/functions/fnc_consume.sqf

```diff
@@ -0,0 +1,67 @@
+/*
+	File Name: fnc_consume.sqf
+	Author: Jak Keen
+*/
+
+#include "../script_component.hpp"
+
+private ["_player", "_mode"];
+
+_player = _this select 1;
+_mode = _this select 2;
+
+switch (_mode) do {
+    case "eat": {
+        private ["_mre"];
+
+        _mre == "";
+
+        {
+            if(_x in (items player)) exitwith {
+                _mre = _x;
+            };
+        } forEach GVAR(food_items);
```

Comment:
you are doing this everywhere.
you can also use the CBA inventoryChanged eventhandler to only check that once on inventory change. Without the inventory changing the contents of the players inventory cannot change anyway.

Reply:
Couldn't find it originally so thought it didn't exist. Now changed